Source/Extension/SafariExtensionHandler.swift

```diff
@@ -19,12 +17,12 @@ class SafariExtensionHandler: SFSafariExtensionHandler {
 
             let basenames = fileBasenames(forHost: host)
 
-            let keysWithValues = Config.FileType.allCases.map { fileType -> (Type, [Name: Content]) in
-                let files: [Name: Content]
+            let keysWithValues = Config.FileType.allCases.map { fileType -> (Type, [[String]]) in
+                let files: [[String]]
```

Comment:
I have to say that the code lost quite a bit of self-documentation.

What do you think of the following?

Keep the type aliases and use `[(Name, Content)]` instead of `[Name: Content]`. Then, only when passing the data to JS we transform the values:

```swift
page.dispatchMessageToScript(
    withName: "onload",
    userInfo: data.mapValues { $0.map(convertTupleToArray) }
)
```

```swift
func convertTupleToArray<T>(_ tuple: (T, T)) -> [T] {
    let (a, b) = tuple
    return [a, b]
}
```

Reply:
For sure, I see your point. It's a trade-off, I did consider this but decided on the more economic but less readable solution of using arrays all the way through. I'll leave the decision to you of course, but I would personally leave it as is I think.

You should feel free to make these edits, I left the PR open to edits so you can commit directly to my fork here if that's convenient.